src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    // i have to pass this prop here because i had some eslint error.
+  }
+
+  hide() {
+    this.hidden = true;
+  };
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  bite(animal) {
+    if (animal.hidden || animal instanceof Carnivore) {
+      return;
+    }
+    animal.health -= 50;
+
+    if (animal.health === 0) {
```

Comment:
Use beast.health <= 0 instead of beast.health === 0 to account for cases when the health goes below 0.